cli/lib/kontena/cli/stacks/list_command.rb

```diff
@@ -91,6 +94,17 @@ def tree_icon(row)
       left_pad + char
     end
 
+    # Converts an array of stack labels into a comma-separated string
+    # or '-' when labels field is not defined.
+    #
+    # @param [Hash] stack
+    # @return [<String>]
+    def stack_labels(stack)
+      labels = (stack['labels'] || ['-']).join(',')
+      # trim labels to fit viewport when exceed 20 chars
+      labels.length > 20 && $stdin.tty? ? "#{labels[0..17]}..." : labels
```

Comment:
Wrap this inside `if $stdin.tty?` (so that something like `stack ls | grep foo=bar` won't fail because of the cutting.

Reply:
@kke, it seems we should be testing the output device `$stdout` instead of input device `$stdin`. Also I don't get why it needs to wrapped inside of `if $stdin.tty?`?

As it is, the logic of `stack_labels` helper:
 - truncate + ellipsis when `lengthy && tty`;
 - don't truncate when `short && tty`;
 - don't truncate when `short && non-tty`;
 - don't truncate when `lengthy && non-tty`;

Is anything else missing? 😄 

Here is a screenshot of the behavior captured as of now (just increased size up to `43`, since `20` is way too short even for a single reasonably-sized label):

<img width="1298" alt="image" src="https://user-images.githubusercontent.com/3822219/37551838-cd2650f2-2965-11e8-8db1-eaa3e44256d0.png">

Reply:
Yes, right, stdout. 

Also missed the use of $stdin.tty? in the existing version for some reason.